servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpScheme.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents HTTP protocol schemes.
+ */
+public final class HttpScheme {
+
+    private static final HttpScheme HTTP = new HttpScheme("http", 80);
+    private static final HttpScheme HTTPS = new HttpScheme("https", 443);
+
+    private final String name;
+    private final int defaultPort;
+
+    private HttpScheme(String name, int defaultPort) {
+        this.name = name;
+        this.defaultPort = defaultPort;
+    }
+
+    /**
+     * Returns the HTTP scheme.
+     *
+     * @return the HTTP scheme
+     */
+    public static HttpScheme http() {
+        return HTTP;
+    }
+
+    /**
+     * Returns the HTTPS scheme.
+     *
+     * @return the HTTPS scheme
+     */
+    public static HttpScheme https() {
+        return HTTPS;
+    }
+
+    /**
+     * Returns the corresponding {@link HttpScheme} for the specified {@code scheme}.
+     *
+     * @param scheme the HTTP protocol scheme in string, case insensitively
+     * @return {@link #http()} if "http", {@link #https()} if "https"
+     * @throws IllegalArgumentException if the specified {@code scheme} is not "http" or "https", case insensitively
+     */
+    public static HttpScheme of(String scheme) {
+        requireNonNull(scheme, "scheme");
+        if (http().name().equalsIgnoreCase(scheme)) {
+            return http();
+        }
+        if (https().name().equalsIgnoreCase(scheme)) {
+            return https();
+        }
+        throw new IllegalArgumentException("Invalid HTTP scheme: " + scheme);
```

Comment:
If in the future we add support for WebSockets, we can definitely add more schemes as constants, but what if users use this HTTP client for other protocols, like WebDAV extensions that use `dav` scheme?